testcases/OpTestDKGSB.py

```diff
@@ -0,0 +1,350 @@
+                        log.info(f"Found {config}={result}")
+                except Exception as e:
+                    if "not set" in str(e):
+                        missing_configs.append(f"{config} (not set)")
+                    else:
+                        log.error(f"Error checking config {config}: {str(e)}")
+                        missing_configs.append(f"{config} (error: {str(e)})")
+            
+            # If any configurations are missing, fail the test with detailed message
+            if missing_configs:
+                error_msg = "Missing or incorrectly configured kernel options:\n"
+                for config in missing_configs:
+                    error_msg += f"- {config}\n"
+                error_msg += "\nPlease ensure these kernel configurations are enabled (y) or built as modules (m)."
+                self.fail(error_msg)
+            
+            log.info("All required kernel configurations are present")
+            
+        except Exception as e:
+            self.fail(f"Error checking kernel configurations: {str(e)}")
+                
+
+    def kernel_grub_signature_check(self):
+        """
+        Check kernel and grub signatures.
+        """
+        try:
+            if not getattr(self, 'util', None):
+                self.util = OpTestUtil(self.conf)
+            self.kernel_signature = self.util.check_kernel_signature()
+            self.grub_filename = self.util.get_grub_file()
+            try:
+                self.grub_signature = self.util.check_grub_signature(self.grub_filename)
+            except Exception as e:
+                log.error("Grub signature check failed for '%s': %s", self.grub_filename, e)
+                self.fail(f"Grub signature check failed for '{self.grub_filename}': {e}")
+            log.info("Kernel signed: %s, Grub file: %s, Grub signed: %s",
+                     self.kernel_signature, self.grub_filename, self.grub_signature)
+            return True
+
+        except Exception as e:
+            log.error("Kernel/Grub signature check failed: %s", e)
+            self.fail(f"Kernel/Grub signature check failed: {e}")
+
+    def execute_hmc_command(self, command):
+        """
+        Execute HMC command and handle errors
+        """
+        self.ensure_hmc_available()
+        try:
+            result = self.cv_HMC.run_command(command)
+            log.info(f"Command executed successfully on HMC: {command}")
+            return result
+        except Exception as e:
+            error_msg = f"Failed to execute HMC command on HMC '{command}': {str(e)}"
+            log.error(error_msg)
+            self.fail(error_msg)
+
+    def ensure_hmc_available(self):
+        """
+        Ensure the HMC object is available.
+        """
+        if not getattr(self, 'cv_HMC', None):
+            msg = (
+                "HMC object is not available (self.cv_HMC is None). "
+            )
+            log.error(msg)
+            self.fail(msg)
```

Comment:
Better, we should move this to the HMC class .. and this can be implicit check before any hmc cmd call